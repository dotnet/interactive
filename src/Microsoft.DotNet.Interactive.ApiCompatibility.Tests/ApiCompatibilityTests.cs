// Copyright (c) .NET Foundation and contributors. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

using System.Diagnostics;
using Assent;
using Microsoft.DotNet.Interactive.CSharp;
using Microsoft.DotNet.Interactive.Documents;
using Microsoft.DotNet.Interactive.Formatting;
using Microsoft.DotNet.Interactive.FSharp;
using Microsoft.DotNet.Interactive.Journey;
using Microsoft.DotNet.Interactive.Kql;
using Microsoft.DotNet.Interactive.PowerShell;
using Microsoft.DotNet.Interactive.SqlServer;
using Microsoft.DotNet.Interactive.Tests.Utility;
using Xunit;

namespace Microsoft.DotNet.Interactive.ApiCompatibility.Tests;

public class ApiCompatibilityTests
{
    private readonly Configuration _configuration;

    public ApiCompatibilityTests()
    {
        _configuration = new Configuration()
                         .SetInteractive(Debugger.IsAttached)
                         .UsingExtension("txt");
    }

    [FactSkipLinux]
    public void Interactive_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Kernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Formatting_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FormatContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Document_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<InteractiveDocument>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void PackageManagement_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PackageRestoreContext>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void Journey_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<Lesson>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void csharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<CSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [Fact(Skip = "need to use signature files")]
    public void fsharp_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<FSharpKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void powershell_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<PowerShellKernel>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void mssql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<MsSqlKernelConnector>();
        this.Assent(contract, _configuration);
    }

    [FactSkipLinux]
    public void kql_api_is_not_changed()
    {
        var contract = ApiContract.GenerateContract<KqlKernelConnector>();
        this.Assent(contract, _configuration);
    }
}