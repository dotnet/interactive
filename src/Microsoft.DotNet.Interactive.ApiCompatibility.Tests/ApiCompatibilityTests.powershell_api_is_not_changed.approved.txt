Microsoft.DotNet.Interactive.PowerShell
  public class PowerShellKernel : Microsoft.DotNet.Interactive.Kernel, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.RequestCompletions>, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.RequestDiagnostics>, Microsoft.DotNet.Interactive.IKernelCommandHandler<Microsoft.DotNet.Interactive.Commands.SubmitCode>, Microsoft.DotNet.Interactive.ValueSharing.ISupportGetValue, Microsoft.DotNet.Interactive.ValueSharing.ISupportSetClrValue, System.IDisposable
    .ctor()
    public System.Func<System.String,System.String> ReadInput { get; set;}
    public System.Func<System.String,Microsoft.DotNet.Interactive.PasswordString> ReadPassword { get; set;}
    public System.Void AddAccelerator(System.String name, System.Type type)
    public System.Void AddModulePath(System.String modulePath)
    public System.Collections.Generic.IReadOnlyCollection<Microsoft.DotNet.Interactive.ValueSharing.KernelValueInfo> GetValueInfos()
    public System.Threading.Tasks.Task HandleAsync(Microsoft.DotNet.Interactive.Commands.SubmitCode submitCode, Microsoft.DotNet.Interactive.KernelInvocationContext context)
    public System.Threading.Tasks.Task HandleAsync(Microsoft.DotNet.Interactive.Commands.RequestCompletions requestCompletions, Microsoft.DotNet.Interactive.KernelInvocationContext context)
    public System.Threading.Tasks.Task HandleAsync(Microsoft.DotNet.Interactive.Commands.RequestDiagnostics requestDiagnostics, Microsoft.DotNet.Interactive.KernelInvocationContext context)
    public System.Threading.Tasks.Task SetValueAsync(System.String name, System.Object value, System.Type declaredType)
    public System.Boolean TryGetValue<T>(System.String name, ref T& value)
  public static class PowerShellKernelExtensions
    public static PowerShellKernel UseProfiles()
Microsoft.DotNet.Interactive.PowerShell.Commands
  public class EnterAzShellCommand : System.Management.Automation.PSCmdlet
    .ctor()
    public System.Guid TenantId { get; set;}
    protected System.Void ProcessRecord()
  public class OutDisplayCommand : System.Management.Automation.PSCmdlet
    .ctor()
    public System.String CustomMimeType { get; set;}
    public System.Object InputObject { get; set;}
    public System.String MimeType { get; set;}
    public System.Management.Automation.SwitchParameter PassThru { get; set;}
    protected System.Void ProcessRecord()
Microsoft.DotNet.Interactive.PowerShell.Host
  public class PSKernelHost : System.Management.Automation.Host.PSHost, System.Management.Automation.Host.IHostSupportsInteractiveSession
    public System.Globalization.CultureInfo CurrentCulture { get;}
    public System.Globalization.CultureInfo CurrentUICulture { get;}
    public System.Guid InstanceId { get;}
    public System.Boolean IsRunspacePushed { get;}
    public System.String Name { get;}
    public System.Management.Automation.PSObject PrivateData { get;}
    public System.Management.Automation.Runspaces.Runspace Runspace { get;}
    public System.Management.Automation.Host.PSHostUserInterface UI { get;}
    public System.Version Version { get;}
    public System.Void EnterNestedPrompt()
    public System.Void ExitNestedPrompt()
    public System.Void NotifyBeginApplication()
    public System.Void NotifyEndApplication()
    public System.Void PopRunspace()
    public System.Void PushRunspace(System.Management.Automation.Runspaces.Runspace runspace)
    public System.Void SetShouldExit(System.Int32 exitCode)
  public class PSKernelHostRawUserInterface : System.Management.Automation.Host.PSHostRawUserInterface
    public System.ConsoleColor BackgroundColor { get; set;}
    public System.Management.Automation.Host.Size BufferSize { get; set;}
    public System.Management.Automation.Host.Coordinates CursorPosition { get; set;}
    public System.Int32 CursorSize { get; set;}
    public System.ConsoleColor ForegroundColor { get; set;}
    public System.Boolean KeyAvailable { get;}
    public System.Management.Automation.Host.Size MaxPhysicalWindowSize { get;}
    public System.Management.Automation.Host.Size MaxWindowSize { get;}
    public System.Management.Automation.Host.Coordinates WindowPosition { get; set;}
    public System.Management.Automation.Host.Size WindowSize { get; set;}
    public System.String WindowTitle { get; set;}
    public System.Void FlushInputBuffer()
    public System.Management.Automation.Host.BufferCell[,] GetBufferContents(System.Management.Automation.Host.Rectangle rectangle)
    public System.Int32 LengthInBufferCells(System.String source, System.Int32 offset)
    public System.Int32 LengthInBufferCells(System.String source)
    public System.Int32 LengthInBufferCells(System.Char c)
    public System.Management.Automation.Host.KeyInfo ReadKey(System.Management.Automation.Host.ReadKeyOptions options)
    public System.Void ScrollBufferContents(System.Management.Automation.Host.Rectangle source, System.Management.Automation.Host.Coordinates destination, System.Management.Automation.Host.Rectangle clip, System.Management.Automation.Host.BufferCell fill)
    public System.Void SetBufferContents(System.Management.Automation.Host.Coordinates origin, System.Management.Automation.Host.BufferCell[,] contents)
    public System.Void SetBufferContents(System.Management.Automation.Host.Rectangle rectangle, System.Management.Automation.Host.BufferCell fill)
  public class PSKernelHostUserInterface : System.Management.Automation.Host.PSHostUserInterface, System.Management.Automation.Host.IHostUISupportsMultipleChoiceSelection
    public System.ConsoleColor DebugBackgroundColor { get; set;}
    public System.ConsoleColor DebugForegroundColor { get; set;}
    public System.ConsoleColor ErrorAccentColor { get; set;}
    public System.ConsoleColor ErrorBackgroundColor { get; set;}
    public System.ConsoleColor ErrorForegroundColor { get; set;}
    public System.ConsoleColor FormatAccentColor { get; set;}
    public System.ConsoleColor ProgressBackgroundColor { get; set;}
    public System.ConsoleColor ProgressForegroundColor { get; set;}
    public System.Management.Automation.Host.PSHostRawUserInterface RawUI { get;}
    public System.Boolean SupportsVirtualTerminal { get;}
    public System.ConsoleColor VerboseBackgroundColor { get; set;}
    public System.ConsoleColor VerboseForegroundColor { get; set;}
    public System.ConsoleColor WarningBackgroundColor { get; set;}
    public System.ConsoleColor WarningForegroundColor { get; set;}
    public System.Collections.Generic.Dictionary<System.String,System.Management.Automation.PSObject> Prompt(System.String caption, System.String message, System.Collections.ObjectModel.Collection<System.Management.Automation.Host.FieldDescription> descriptions)
    public System.Int32 PromptForChoice(System.String caption, System.String message, System.Collections.ObjectModel.Collection<System.Management.Automation.Host.ChoiceDescription> choices, System.Int32 defaultChoice)
    public System.Collections.ObjectModel.Collection<System.Int32> PromptForChoice(System.String caption, System.String message, System.Collections.ObjectModel.Collection<System.Management.Automation.Host.ChoiceDescription> choices, System.Collections.Generic.IEnumerable<System.Int32> defaultChoices)
    public System.Management.Automation.PSCredential PromptForCredential(System.String caption, System.String message, System.String userName, System.String targetName)
    public System.Management.Automation.PSCredential PromptForCredential(System.String caption, System.String message, System.String userName, System.String targetName, System.Management.Automation.PSCredentialTypes allowedCredentialTypes, System.Management.Automation.PSCredentialUIOptions options)
    public System.String ReadLine()
    public System.Security.SecureString ReadLineAsSecureString()
    public System.Void Write(System.ConsoleColor foregroundColor, System.ConsoleColor backgroundColor, System.String value)
    public System.Void Write(System.String value)
    public System.Void WriteDebugLine(System.String message)
    public System.Void WriteErrorLine(System.String value)
    public System.Void WriteInformation(System.Management.Automation.InformationRecord record)
    public System.Void WriteLine()
    public System.Void WriteLine(System.ConsoleColor foregroundColor, System.ConsoleColor backgroundColor, System.String value)
    public System.Void WriteLine(System.String value)
    public System.Void WriteProgress(System.Int64 sourceId, System.Management.Automation.ProgressRecord record)
    public System.Void WriteVerboseLine(System.String message)
    public System.Void WriteWarningLine(System.String message)
